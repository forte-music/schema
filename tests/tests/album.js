import gql from 'graphql-tag';
import { print } from 'graphql/language/printer';

import AlbumFields from './fragments/AlbumFields.graphql';
import ArtistFields from './fragments/ArtistFields.graphql';
import SongFields from './fragments/SongFields.graphql';

import client from '../client';

const variables = { albumId: 'album:4' };
it('should get an album by id', async () => {
  const query = gql`
    query($albumId: ID!) {
      album(id: $albumId) {
        ...AlbumFields

        artist {
          ...ArtistFields
        }

        songs {
          ...SongFields
        }
      }
    }

    ${AlbumFields}
    ${ArtistFields}
    ${SongFields}
  `;

  const { album } = await client.request(print(query), variables);

  expect(album).toMatchSnapshot();
});

it('should calculate the duration for an album', async () => {
  const { album } = await client.request(
    `
      query($albumId: ID!) {
        album(id: $albumId) {
          id
          duration
        }
      }
    `,
    variables
  );

  expect(album).toMatchObject({
    id: variables.albumId,
    duration: 74 * 60 + 22,
  });
});
