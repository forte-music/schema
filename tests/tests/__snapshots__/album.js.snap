// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get an album by id 1`] = `
Object {
  "artist": Object {
    "id": "artist:4",
    "name": "G-Eazy",
  },
  "artworkUrl": "https://upload.wikimedia.org/wikipedia/en/3/31/G-Eazy_-_The_Beautiful_%26_Damned.png",
  "duration": 4462,
  "id": "album:4",
  "name": "The Beautiful & Damned",
  "releaseYear": 2017,
  "songs": Array [
    Object {
      "diskNumber": 1,
      "id": "song:4:5",
      "name": "Sober",
      "streamUrl": "",
      "trackNumber": 5,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:2",
      "name": "Pray for Me",
      "streamUrl": "",
      "trackNumber": 2,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:1",
      "name": "The Beautiful & Damned",
      "streamUrl": "",
      "trackNumber": 1,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:13",
      "name": "Crash & Burn",
      "streamUrl": "",
      "trackNumber": 13,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:16",
      "name": "No Less",
      "streamUrl": "",
      "trackNumber": 16,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:3",
      "name": "Him & I",
      "streamUrl": "",
      "trackNumber": 3,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:9",
      "name": "That's a Lot",
      "streamUrl": "",
      "trackNumber": 9,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:11",
      "name": "Gotdamn",
      "streamUrl": "",
      "trackNumber": 11,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:4",
      "name": "But a Dream",
      "streamUrl": "",
      "trackNumber": 4,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:7",
      "name": "No Limit",
      "streamUrl": "",
      "trackNumber": 7,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:6",
      "name": "Legend",
      "streamUrl": "",
      "trackNumber": 6,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:14",
      "name": "Summer in December",
      "streamUrl": "",
      "trackNumber": 14,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:10",
      "name": "Pick Me Up",
      "streamUrl": "",
      "trackNumber": 10,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:20",
      "name": "Eazy",
      "streamUrl": "",
      "trackNumber": 20,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:19",
      "name": "Love Is Gone",
      "streamUrl": "",
      "trackNumber": 19,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:17",
      "name": "Mama Always Told Me",
      "streamUrl": "",
      "trackNumber": 17,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:18",
      "name": "Fly Away",
      "streamUrl": "",
      "trackNumber": 18,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:15",
      "name": "Charles Brown",
      "streamUrl": "",
      "trackNumber": 15,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:8",
      "name": "The Plan",
      "streamUrl": "",
      "trackNumber": 8,
    },
    Object {
      "diskNumber": 1,
      "id": "song:4:12",
      "name": "Leviathan",
      "streamUrl": "",
      "trackNumber": 12,
    },
  ],
}
`;
